src/main/java/com/roommatematcher/enterprise/dao/ProfileDAO.java

```diff
@@ -21,4 +24,11 @@
         List<Profile> profiles = execute.body();
         return profiles;
     }
+
+    //this is an implementation of the getProfileId() method that we implemented in ProfileDTO
+    public Profile save(Profile profile) throws Exception {
+        Integer profileID = Integer.parseInt(profile.getProfileId());
+        allProfiles.put(profileID, profile);
+        return profile;
```

Comment:
It looks like you are storing to the allProfiles collection, but never retrieving from it.  Bring it full circle, and fetch from it in the fetch method.